astarte-device-sdk/AstarteDeviceSDK.cpp

```diff
@@ -302,7 +339,9 @@ bool AstarteDeviceSDK::sendData(const QByteArray &interface, const QVariantHash
     }
     // Verify mappings
     QHash< QByteArray, QVariant::Type > mappingToType = m_producers.value(interface)->mappingToType();
-    if (mappingToType.size() != value.size()) {
+    QHash< QByteArray, QVariant::Type > mappingToArrayType = m_producers.value(interface)->mappingToArrayType();
```

Comment:
honestly I cannot understand what you are doing here. Can you explain it?

Reply:
Array types are in a different QHash than the scalar types to not mix them